packages/aws-cdk-lib/aws-batch/lib/job-queue.ts

```diff
@@ -135,6 +143,79 @@ export interface OrderedComputeEnvironment {
   readonly order: number;
 }
 
+/**
+ * Specifies an action that AWS Batch will take after the job has remained at
+ * the head of the queue in the specified state for longer than the specified time.
+ */
+
+export interface JobStateTimeLimitAction {
+  /**
+ * The action to take when a job is at the head of the job queue in the specified state
+ * for the specified period of time.
+ */
+  readonly action: JobStateTimeLimitActionsAction;
+
+  /**
+   * The approximate amount of time, in seconds, that must pass with the job in the specified
+   * state before the action is taken.
+   * The minimum value is 600 (10 minutes) and the maximum value is 86,400 (24 hours).
+   */
+  readonly maxTimeSeconds: Duration;
+
+  /**
+   * The reason to log for the action being taken.
+   * @see https://docs.aws.amazon.com/batch/latest/userguide/troubleshooting.html#job_stuck_in_runnable
+   */
+  readonly reason: JobStateTimeLimitActionsReason;
+
+  /**
+   * The state of the job needed to trigger the action.
+   */
+  readonly state: JobStateTimeLimitActionsState;
```

Comment:
This is similar comment above, it may be better to make `state` optional and set default value to `JobStateTimeLimitActionsState.RUNNABLE`.

Reply:
Thanks.

I have changed `state` to be optional and set its default value to `JobStateTimeLimitActionsState.RUNNABLE`.